src/averageAges.js

```diff
@@ -54,8 +76,26 @@ function calculateWomenAverageAge(people, withChildren) {
  *
  * @return {number}
  */
+
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // write code here
+  const ageDifferences = onlyWithSon
+    ? people.map(person => {
+      const motherOfPerson = person.sex === 'm'
+        ? people.find(human => human.name === person.mother)
+        : undefined;
```

Comment:
Don't use the ternary operator if you have one condition and you don't need to do anything if the condition is false.

Reply:
correct!